src/wrap-store/shallowDiff.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Returns a new Object containing only the fields in `new` that differ from `old`
+ *
+ * @param {Object} old
+ * @param {Object} new
+ * @return {Object{updated, removed}} An object containing the updated and removed
+ *   fields. Updated is an object containing any fields added or changed. Removed
+ *   is an array of keys that exist in the old object but are missing in the new object.
+ */
+export default function shallowDiff(oldObj, newObj) {
+  const updated = {};
```

Comment:
Maybe we could simplify the `diff` to just be an array of tuples that represent changes (updates and removals both count) or something? 

We could then make `removed` and `updated` "computed" properties by implementing them as getters.

Idk just an idea :)